cot-macros/src/lib.rs

```diff
@@ -192,3 +192,82 @@ pub(crate) fn cot_ident() -> proc_macro2::TokenStream {
+        &data_struct.fields
+    } else {
+        let err = Error::custom("Only structs can derive `FromRequestParts`");
+        return err.write_errors();
+    };
+
+    match fields {
+        Fields::Named(fields_named) => {
+            let field_initializers = fields_named.named.iter().map(|field: &Field| {
+                let field_name = &field.ident;
+                let field_type = &field.ty;
+                quote! {
+                    #field_name: #field_type::from_request_parts(parts).await?,
+                }
+            });
+
+            quote! {
+            #[::core::automatically_derived]
+                            impl
+                            impl #cot::axum::extract::FromRequestParts<#cot::http::Request, #cot::anyhow::Error> for #struct_name {
+                                async fn from_request_parts(
+                                    parts: &mut #cot::axum::extract::RequestParts,
+                                ) -> ::std::result::Result<Self, #cot::anyhow::Error> {
+                                    Ok(Self {
+                                        #(#field_initializers)*
+                                    })
+                                }
+                            }
+                        }
+        }
+        Fields::Unnamed(fields_unnamed) => {
+            let field_initializers = fields_unnamed.unnamed.iter().map(|field: &Field| {
+                let field_type = &field.ty;
+                quote! {
+                    #field_type::from_request_parts(parts).await?,
+                }
+            });
+
+            quote! {
+            #[::core::automatically_derived]
+                            impl
+                            impl #cot::axum::extract::FromRequestParts<#cot::http::Request, #cot::anyhow::Error> for #struct_name {
+                                async fn from_request_parts(
+                                    parts: &mut #cot::axum::extract::RequestParts,
+                                ) -> ::std::result::Result<Self, #cot::anyhow::Error> {
+                                    Ok(Self(
+                                        #(#field_initializers)*
+                                    ))
+                                }
+                            }
+                        }
+        }
+        Fields::Unit => {
+            quote! {
+            #[::core::automatically_derived]
+                            impl
+                            impl #cot::axum::extract::FromRequestParts<#cot::http::Request, #cot::anyhow::Error> for #struct_name {
+                                async fn from_request_parts(
+                                    parts: &mut #cot::axum::extract::RequestParts,
+                                ) -> ::std::result::Result<Self, #cot::anyhow::Error> {
+                                    Ok(Self)
+                                }
+                            }
+                        }
+        }
+    }
```

Comment:
Please see my comment above.

Also, can we unify the `match` blocks a little? Seems like the only thing that is changed between them is the `Self` construction; this should be easy to extract out of the `match` which would result in less code duplication.